src/transform/strategies/date_to_age.rs

```diff
@@ -0,0 +1,374 @@
+        let dob_table = tables.iter().find(|table|                !table
+            .filter_columns()
+            .where_data_context(Filter::Is(&Context::DateOfBirth))
+            .collect()
+            .is_empty()).expect("Unexpectedly could not find table with DateOfBirth data when applying DateToAge strategy.");
+
+        let dob_columns = dob_table
+            .filter_columns()
+            .where_data_context(Filter::Is(&Context::DateOfBirth))
+            .collect();
+
+        let dob_col_name = dob_columns
+            .first()
+            .expect("Unexpectedly could not find DateOfBirth column in table.")
+            .name()
+            .as_str();
+
+        Ok(dob_table.create_subject_id_string_data_hash_map(dob_col_name)?)
+    }
+
+    /// Given the date of birth of a patient, and a date in their life
+    /// this will calculate the age of a patient at that date.
+    ///
+    /// An error will be thrown if the date of birth, or the date, cannot be interpreted as
+    /// chrono::NaiveDate.
+    fn date_and_dob_to_age(dob: String, date: &str) -> Result<String, StrategyError> {
+        let dob_object = dob.parse::<NaiveDate>()?.and_hms_opt(0, 0, 0).unwrap();
+        let date_object = date.parse::<NaiveDate>()?.and_hms_opt(0, 0, 0).unwrap();
+        let diff = date_diff(dob_object, date_object);
+        let dur = Duration::new(
+            diff.years as f32,
+            diff.months as f32,
+            diff.days as f32,
+            0f32,
+            0f32,
+            0f32,
+        );
+        Ok(dur.to_string())
+    }
+
+    fn upsert_mapping_error(
+        error_info: &mut HashSet<MappingErrorInfo>,
+        date_col_name: &str,
+        table: &ContextualizedDataFrame,
+        date: &str,
+    ) {
+        let mapping_error_info = MappingErrorInfo {
+            column: date_col_name.to_string(),
+            table: table.context().name().to_string(),
+            old_value: date.to_string(),
+            possible_mappings: vec![],
+        };
+        if !error_info.contains(&mapping_error_info) {
+            error_info.insert(mapping_error_info);
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::config::TableContext;
+    use crate::config::context::AGE_CONTEXTS;
+    use crate::config::table_context::Identifier::Regex;
+    use crate::config::table_context::SeriesContext;
+    use polars::df;
+    use polars::frame::DataFrame;
+    use rstest::{fixture, rstest};
+
+    #[fixture]
+    fn dob_alice() -> String {
+        "1995-06-01".to_string()
+    }
+
+    #[fixture]
+    fn dob_bob() -> String {
+        "1990-12-01".to_string()
+    }
+
+    #[fixture]
+    fn dob_charlie() -> String {
+        "1980-01-08".to_string()
+    }
+
+    #[fixture]
+    fn onset_bob() -> String {
+        "1991-12-01".to_string()
+    }
+
+    #[fixture]
+    fn onset_charlie() -> String {
+        "2025-11-25".to_string()
+    }
+
+    #[fixture]
+    fn df1() -> DataFrame {
```

Comment:
Whats the difference between df1 and df2? Let their names reflect that.

Same for tc1 and tc2